src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Resource.Designer.targets

```diff
@@ -103,6 +104,7 @@ Copyright (C) 2016 Xamarin. All rights reserved.
   <GenerateResourceDesignerIntermediateClass
       IsApplication="$(AndroidApplication)"
       Namespace="$(AndroidResgenNamespace)"
+      Modifier="$(AndroidResourceDesignerClassModifier)"
```

Comment:
If I set `-p:AndroidResourceDesignerClassModifier=foo`, would something give a decent error message? Or would we basically just get a C# error message?

Reply:
Add the XA1038 for this issue